### ISSUE TITLE
Malformed XMTok

### ISSUE THREAD

**Issue by jbs1** [jbs1/tract2#161]
_migrated from Trac, where originally posted by **kpj** on 10-Jun-2013 4:00pm_

The file `m1242256/m1242256.tex` raises
```
Error:malformed:ltx:XMTok <ltx:XMTok> isn't allowed here
  Anonymous String#textrange(from=0;1,to=0;0)
  In Document[@0x2d8caa8] Anonymous String#textrange(from=0;1,to=0;0)
```
MK suggested that this might be caused by the use of \textusperscript inside \unit{}.

\unit{} is defined as
```
RawTeX('
\@ifundefined{unit}{%
   \DeclareRobustCommand*{\unit}[1]{%
      \ensuremath{\def\mu{\mbox{\textmu}}\def~{\,}%
      \unskip~%
      \ifx\testbx\f@series\mathbf{#1}\else\mathrm{#1}\fi}}}%
   {}

');
```

Any suggestions?

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **miller** on 11-Jun-2013 11:59pm_

Not without some hint as to what markup is involved.

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **kpj** on 12-Jun-2013 8:16am_

I added an example which creates the error.

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **miller** on 12-Jun-2013 3:40pm_

Aha, how silly! I'd defined `\textsuperscript` using a text
`ltx:sup` element in order to avoid math entirely,
but I hadn't forced it process its argument in text mode.

I hadn't imagined someone would use textsuperscript in math mode!?!?!

Anyway; fixed now! Thanks for the report

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **kohlhase** on 12-Jun-2013 3:43pm_

maybe this also fixes #1508

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **kpj** on 13-Jun-2013 9:47am_

Yep, this one is fixed now, thanks a lot!
